Sources/Swift/Tools/Batcher/InMemoryBatchStorage.swift

```diff
@@ -0,0 +1,25 @@
+struct InMemoryBatchStorage<Element: Encodable>: BatchStorage {
+    private var elements: [Data] = []
+    var size: Int = 0
+
+    init() {}
+
+    mutating func append(_ element: Element) throws {
+        let encoded = try JSONEncoder().encode(element)
+        elements.append(encoded)
+        size += encoded.count
+    }
+
+    mutating func flush() {
+        elements.removeAll()
+        size = 0
+    }
```

Comment:
`m`: The `flush` method name is a bit misleading. This method doesn't flush; it clears the storage. Either we rename it to something like `clear` or we put the flushing logic in here.